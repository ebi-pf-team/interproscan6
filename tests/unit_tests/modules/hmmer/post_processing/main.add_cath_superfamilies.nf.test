nextflow_process {

    name "Test Process ADD_CATH_SUPERFAMILIES"
    script "interproscan/modules/hmmer/post_processing/main.nf"
    process "ADD_CATH_SUPERFAMILIES"

    stage {
        symlink "tests/unit_tests/test_inputs/hmmer_post_processing/cath/cath_resolve_out"
    }

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                input[0] = file("meta/tests/unit_tests/test_inputs/hmmer_post_processing/cath/cath_resolve_out")
                input[1] = [
                    "param0",
                    "cath_resolve_switches",
                    "model2sf_map",
                    "discontinuous_regs"
                ]
                input[2] = true
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
