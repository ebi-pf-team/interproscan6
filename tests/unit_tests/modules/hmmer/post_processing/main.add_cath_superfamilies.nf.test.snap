{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "cath_resolve_out.cath_superfamilies:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-09-23T11:45:51.16754"
    }
}